parlai/agents/transformer/polyencoder.py

```diff
@@ -86,7 +96,8 @@ def __init__(self, opt, shared=None):
             self.model = torch.nn.DataParallel(self.model)
 
     def build_model(self, states=None):
-        return PolyEncoderModule(self.opt, self.dict, self.NULL_IDX)
+        self.model = PolyEncoderModule(self.opt, self.dict, self.NULL_IDX)
+        return self.model
```

Comment:
i don't think you need to set self.model here, since the call to build_model is like `self.model = self.build_model(...)`, so this could just be `return  PolyEncoderModule(self.opt, self.dict, self.NULL_IDX)`